### PR TITLE
Add action that updates api.grafbase.com schema

### PR DESCRIPTION
**Pull request by obmarg**
# Description

We have a local cached copy of the `api.grafbase.com` schema in this repository,  which we use to validate some queries we make against that API.  Currently it's a manual task to update this.

This PR adds a daily GitHub action to handle updates instead - it'll raise a PR if there are any changes, otherwise it'll do nothing.

# Type of change

- [ ] 💔 Breaking
- [ ] 🚀 Feature
- [ ] 🐛 Fix
- [x] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [ ] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: cli/.prettierignore [hunk 1]
```diff
@@ -1,4 +1,6 @@
 target
 crates/cli/tests/graphql/compilation_error/schema.graphql
+crates/cli/tests/graphql/compilation_error/schema.graphql
```
Comment (obmarg) [inline cli/.prettierignore:3]:
```suggestion
```

File: .github/workflows/update-grafbase-schema.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: Check grafbase.com for GraphQL schema updates
+on: 
+  workflow_dispatch:
+  schedule:
+    - cron: 0 9 * * *
```
Comment (obmarg) [inline .github/workflows/update-grafbase-schema.yml:5]:
```suggestion
  # Will uncomment schedule when I'm sure this works
  #schedule:
  #  - cron: 0 9 * * *
```

**Review by Finistere — APPROVED** [review]
Nice!

File: .github/workflows/update-grafbase-schema.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+jobs:
+  update-grafbase-schema:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      pull-requests: write
+    steps:
+    - uses: actions/checkout@v3
+    - name: Run introspection
+      uses: obmarg/action-graphql-introspect@main
```
Comment (jakubadamw) [inline .github/workflows/update-grafbase-schema.yml:16]: We use `rover` for this in some other jobs. Could we stick to one tool, please?

**Pull request closed without merging**